### DISCUSSION TITLE
[Question] Get highest ASN via API

### DISCUSSION THREAD

**Discussion by Bart1909** [jonaswinkler/paperless-ng#518] [General]
Hey,
first of all, thank you very much for this nice product!

I'm not at the point, that I have scanned all physical documents and imported them into paperless-ng.
Now the highest ASN is, lets say, 505.

If I get new documents, I want to write the next ASN (506) on the document and scan it. Afterwards, the note is not necessarily sorted directly, but perhaps first put aside and later sorted with other documents.

It would be nice, if there is an API method to get the highest ASN in the system, so I can show this in our home automation system and see, what number I have to write on the document, without looking into the documents or paperless-ng.

Is there something existing already or would it possible, to implement this?

Best wishes

Tobi

**Comment by jonaswinkler** [comment]
Hello!

Please use the discussion section for requests like this. The issue board should really only track issues with the software.

Regarding your request: You could perform a request on the API as follows:

`https://paperless/api/documents/?archive_serial_number__isnull=false&ordering=-archive_serial_number`

This will only list documents that have an ASN assigned and the first document returned will be the document with the highest ASN. The rest is parsing that result in your automation system.

You can also perform that in your browser - paperless comes with a generated browsable API interface.

Regarding Authentication: The API supports basic auth, token auth (see documentation, API section), and session auth (session ID in cookies).

**Reply by Bart1909** [reply]
Sorry for posting this as an issue! 

And thank you for the url! This works fine :-)
